From bf53c92c93c38abc6d98cc9ba405e905e8a2795d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 16 May 2016 23:06:04 -0700
Subject: [PATCH] Add support for nios2

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/shared/architecture.c | 2 ++
 src/shared/architecture.h | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/src/shared/architecture.c b/src/shared/architecture.c
index a9ecfc1..4c46835 100644
--- a/src/shared/architecture.c
+++ b/src/shared/architecture.c
@@ -121,6 +121,8 @@ int uname_architecture(void) {
                 { "tilegx",     ARCHITECTURE_TILEGX   },
 #elif defined(__cris__)
                 { "crisv32",    ARCHITECTURE_CRIS     },
+#elif defined(__nios2__)
+                { "nios2",      ARCHITECTURE_NIOS2    },
 #else
 #error "Please register your architecture here!"
 #endif
diff --git a/src/shared/architecture.h b/src/shared/architecture.h
index 26679e2..7a4cbf3 100644
--- a/src/shared/architecture.h
+++ b/src/shared/architecture.h
@@ -57,6 +57,7 @@ enum {
         ARCHITECTURE_M68K,
         ARCHITECTURE_TILEGX,
         ARCHITECTURE_CRIS,
+        ARCHITECTURE_NIOS2,
         _ARCHITECTURE_MAX,
         _ARCHITECTURE_INVALID = -1
 };
@@ -194,6 +195,10 @@ int uname_architecture(void);
 #elif defined(__cris__)
 #  define native_architecture() ARCHITECTURE_CRIS
 #  error "Missing LIB_ARCH_TUPLE for CRIS"
+#elif defined(__nios2__)
+#  define native_architecture() ARCHITECTURE_NIOS2
+#  define LIB_ARCH_TUPLE "nios2-linux-gnu"
+#  define PROC_CPUINFO_MODEL "cpu"
 #else
 #  error "Please register your architecture here!"
 #endif
-- 
2.8.2

